src/plugins/accounting.cpp

```diff
@@ -103,10 +103,71 @@ json_t* Association::to_json () const
         }
     }
 
+    json_t *hj_json = json_object ();
+    if (!hj_json) {
+        json_decref (user_queues);
+        json_decref (user_projects);
+        json_decref (queue_usage_json);
+        return nullptr;
+    }
+
+    for (const auto &entry : hj) {
+        const Job &job = entry.second;
+        json_t *job_json = json_pack ("{s:i, s:i, s:s, s:o}",
+                                      "nnodes", job.nnodes,
+                                      "ncores", job.ncores,
+                                      "queue", job.queue.c_str (),
+                                      "deps", json_array ());
+
+        if (!job_json) {
+            json_decref (user_queues);
+            json_decref (user_projects);
+            json_decref (queue_usage_json);
+            json_decref (hj_json);
+            return nullptr;
+        }
```

Comment:
I think you could reduce a bunch of these duplicated `json_decref` blocks if you put them all at the end of the function in a `goto error` type thing. I think you would need to initialize all of the variables to `nullptr` at the beginning of the function though.

```C
if (!job_json) {
    goto error;
}

...

error:
    json_decref (user_queues);
    json_decref (user_projects);
    ...
    return nullptr;
```

Reply:
These are great suggestions, thanks. I kind of hate the duplicated code to `json_decref` every object as the function progresses, so just creating an `error` label makes more sense. Should I lump this fix in with the same commit that adds the new object to this method? Or make it a separate commit?

Reply:
I'm leaning towards making this just a separate commit if that is okay.

Reply:
yeah that's fine!